cli/azd/cmd/mcp.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/alpha"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/llm/tools"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func newMcpFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *mcpFlags {
+	flags := &mcpFlags{}
+	flags.Bind(cmd.Flags(), global)
+
+	return flags
+}
+
+func newMcpCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "mcp",
+		Short: "Run MCP server.",
+	}
+}
+
+type mcpFlags struct {
+	global *internal.GlobalCommandOptions
+	internal.EnvFlag
+}
+
+func (i *mcpFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	i.EnvFlag.Bind(local, global)
+	i.global = global
+}
```

Comment:
How is the MCP command going to use the environment flag?

Reply:
We haven't decide if:
- You need to start the mcp server with --cwd and optionally a env name so the server is scoped to an specific project
or
- Each tool needs to get the path and env name from the mcp-client (agent must get it and send it all the time)
or
- Tools should never depend on azd specific paths and should do and return work which the agent can apply to the current project.

Depending on that, we would either remove this env or do something else.
Right now I just put it there as I was copy/pasting from another command XD